### DISCUSSION TITLE
Using PlotJuggler to translate from .ulg to CSV

### DISCUSSION THREAD

**Discussion by CharlieRigel** [facontidavide/PlotJuggler#495] [Q&A]
Hey, I have been working a bit with PlotJuggler recently and I noticed the functionality of being able to take in a .ulg file and then export it into a CSV file. Is there any way to incorporate this functionality into my own code? or is there a peice of the code that does this you can point me to? I am working on a project to analyze some data that is in a .ulg format and I need to translate it into a CSV file first. I have pyulog but their ulg to csv program splits it into about 50 seperate documents but PlotJuggler does it very nicely into one document and I was hoping there was some way to access that functionality without having to do it manually. Any help would be amazing!

**Comment by facontidavide** [comment]
The only thing I can suggest is to have a look at the code.

On one side, we have the ULOG plugin that convert the file to **PJ::PlotDataMapRef**.
The CSV exporter takes this data structuure and creates tge CSV.

It should not be a huge amount of work to extract the code, but it requires good understanding of C++.
